### PR TITLE
test(cosmic-swingset): fix Rosetta flake

### PR DESCRIPTION
**Pull request by michaelfig**
Make the Rosetta cosmic-swingset test more robust, by avoiding block timeouts and instead just running to completion or first error.

Also works around `ag-chain-cosmos export` that broke under MacOS/`ag-chain-cosmos` (fused brain), by using `agd export` directly (half-brain).

### CODE REVIEW

File: packages/cosmic-swingset/test/test-make.js [hunk 1]
```diff
@@ -1,23 +1,33 @@
...
+  const pspawnAgd = pspawn('../../bin/agd', io);
   const scenario2 = makeScenario2({ pspawnMake, pspawnAgd, log: t.log });
-  await scenario2.setup();
+  return { scenario2, pspawnAgd, pspawnMake };
+};
 
-  t.context = { scenario2, pspawnAgd };
+/**
+ * @param {import('ava').ExecutionContext<Awaited<ReturnType<makeTestContext>>>} t
+ */
```
Comment (dckc) [inline packages/cosmic-swingset/test/test-make.js]: this type should be inferred now, no?

File: packages/cosmic-swingset/test/test-make.js [hunk 2]
```diff
@@ -45,18 +55,21 @@ test.serial('make and exec', async t => {
...
 
-  const queryGrace = 6; // time to query state before shutting down
-  const [_run] = await Promise.all([
-    scenario2.runToHalt({
-      BLOCKS_TO_RUN: enoughBlocksToProvision + queryGrace,
-    }),
-    scenario2.runRosettaCI({}),
+  const code = await Promise.race([
+    rosetta.exit,
+    runChain.exit.then(c => `chain exited unexpectedly with code ${c}`),
```
Comment (dckc) [inline packages/cosmic-swingset/test/test-make.js]: shouldn't this fail the test? or is it too unreliable?

**Review by dckc — APPROVED** [review]
Looks worth a try, though...

I'm not clear on what's our diagnosis of the problem. What's our theory on what's going wrong?

File: packages/cosmic-swingset/test/test-make.js (hunk 2 above)
Comment (michaelfig) [reply packages/cosmic-swingset/test/test-make.js]: I added a comment explaining that we don't want an unhandled rejection if the chain loses the race (normal shutdown behaviour).

**Comment by michaelfig** [comment]
> I'm not clear on what's our diagnosis of the problem. What's our theory on what's going wrong?

It looks like `rosetta-ci.sh`'s interaction with the chain is nondeterministic, and sometimes takes longer than other times.  Then the chain would halt before `rosetta-ci.sh` was done with it, bringing down the whole test.  My change is to run the chain until `rosetta-ci.sh` really exits (or the chain fails).

**Pull request merged**